packages/usdk/packages/upstreet-agent/packages/react-agents/classes/discord-manager.ts

```diff
@@ -411,11 +466,71 @@ export class DiscordBot extends EventTarget {
       });
     };
 
+    // message reactions
+    const _bindIncomingMessageReactions = () => {
+      const handleReaction = (e: MessageEvent, eventType: string) => {
+        const {
+          userId,
+          messageId,
+          emoji,
+          channelId,
+          userDisplayName,
+        } = e.data;
+
+        console.log(eventType, {
+          userId,
+          userDisplayName,
+          messageId,
+          emoji,
+          channelId,
+        });
+
+        // look up conversation
+        const conversation = this.dmConversations.has(userId)
```

Comment:
I'm not sure if this is right. I think the DM-ness of the message is supposed to be indicated by whether there is a `channelId`, not by whether we happen to have a DM with the `userId`.

Reply:
Updated.
Using the discord bot's guildId extracted from the message. guildId for dm's is always null and is not no null for messages sent in a server.
Overall though, im not sure why we have 2 seperate arrays (dmChannels and textChannels), for dm's we can createDM to get the channelId and send over ws and only maintain 1 channels array. For userWhitelisting we can use the discord "username" and whitelist server side